server/webapp/WEB-INF/rails/webpack/views/pages/pipelines/parameters_editor.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {MithrilViewComponent} from "jsx/mithril-component";
+import _ from "lodash";
+import m from "mithril";
+import Stream from "mithril/stream";
+import {PipelineParameters} from "models/pipeline_configs/parameters";
+import * as Buttons from "views/components/buttons";
+import {IdentifierInputField} from "views/components/forms/common_validating_inputs";
+import {Form, FormBody} from "views/components/forms/form";
+import {TextField} from "views/components/forms/input_fields";
+import {Table} from "views/components/table";
+import * as Tooltip from "views/components/tooltip";
+import {TooltipSize} from "views/components/tooltip";
+import css from "./environment_variables_editor.scss";
+
+interface Attrs {
+  parameters: Stream<PipelineParameters[]>;
+}
+
+export class PipelineParametersEditor extends MithrilViewComponent<Attrs> {
+  paramList: Stream<PipelineParameters[]> = Stream([new PipelineParameters("", "")]);
```

Comment:
Here, you don't have to specify an explicit type if you don't want to. TS has enough info to infer the type, so you can simplify this to:

```typescript
paramList = Stream([new PipelineParameters("", "")]);
```